test/e2e/cloud-provider-azure.go

```diff
@@ -24,49 +24,30 @@ import (
 	"fmt"
 
 	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-	"k8s.io/apimachinery/pkg/labels"
-	"sigs.k8s.io/cluster-api/test/framework"
 	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	cloudProviderAzureHelmRepoURL    = "https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo"
-	cloudProviderAzureChartName      = "cloud-provider-azure"
-	cloudProvierAzureHelmReleaseName = "cloud-provider-azure-oot"
+	cloudProviderAzureHelmRepoURL     = "https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo"
+	cloudProviderAzureChartName       = "cloud-provider-azure"
+	cloudProviderAzureHelmReleaseName = "cloud-provider-azure-oot"
 )
 
 // InstallCloudProviderAzureHelmChart installs the official cloud-provider-azure helm chart
-// Fulfills the clusterctl.Waiter type so that it can be used as ApplyClusterTemplateAndWaitInput data
-// in the flow of a clusterctl.ApplyClusterTemplateAndWait E2E test scenario
-func InstallCloudProviderAzureHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, result *clusterctl.ApplyClusterTemplateAndWaitResult) {
-	By(fmt.Sprintf("Ensuring the kubeconfig secret for cluster %s/%s exists before installing cloud-provider-azure components", input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName))
-	WaitForWorkloadClusterKubeconfigSecret(ctx, input)
+// and validates that expected pods exist and are Ready
```

Comment:
```suggestion
// and validates that expected pods exist and are Ready.
```

Reply:
Obviously not the end of the world, but I thought we had a linter in place that enforced "comments should end with a period." I'll investigate.